libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -251,8 +248,13 @@ void AP_Proximity_RPLidarA2::get_readings()
     uint32_t nbytes = _uart->available();
 
     while (nbytes-- > 0) {
-
-        uint8_t c = _uart->read();
+        int16_t readChar = _uart->read();
+        uint8_t c;
+        if (readChar < 0) {
+            continue;
```

Comment:
Is it just me or is this an infinite loop if the uart dies?

Reply:
https://github.com/ArduPilot/ardupilot/blob/4c4adc2f42d7d7a40d3d105dc4442a76fd7e51b3/libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp#L250
Since it counts down in this line, it does not perform infinite loop.

Reply:
The type of nbytes must be changed to int IMHO.

Reply:
@amilcarlucas san..
I get out of the loop if nbytes becomes 0.
I think that it is good to take into consideration the case where it does not become 0 for some reason.